rpc/handlers_test.go

```diff
@@ -1494,6 +1494,123 @@ func TestTransactionReceiptByHash(t *testing.T) {
 	})
 }
 
+func TestBlockWithReceipts(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+
+	mockReader := mocks.NewMockReader(mockCtrl)
+	handler := rpc.New(mockReader, nil, nil, "", nil)
+
+	t.Run("transaction not found", func(t *testing.T) {
+		blockID := rpc.BlockID{Number: 777}
+
+		mockReader.EXPECT().BlockByNumber(blockID.Number).Return(nil, db.ErrKeyNotFound)
+
+		resp, rpcErr := handler.BlockWithReceipts(blockID)
+		assert.Nil(t, resp)
+		assert.Equal(t, rpc.ErrBlockNotFound, rpcErr)
+	})
+	t.Run("l1head failure", func(t *testing.T) {
+		blockID := rpc.BlockID{Number: 777}
+		block := &core.Block{
+			Header: &core.Header{},
+		}
+
+		err := errors.New("l1 failure")
+		mockReader.EXPECT().BlockByNumber(blockID.Number).Return(block, nil)
+		mockReader.EXPECT().L1Head().Return(nil, err)
+
+		resp, rpcErr := handler.BlockWithReceipts(blockID)
+		assert.Nil(t, resp)
+		assert.Equal(t, rpc.ErrInternal.CloneWithData(err.Error()), rpcErr)
+	})
+
+	client := feeder.NewTestClient(t, &utils.Mainnet)
+	mainnetGw := adaptfeeder.New(client)
+
+	t.Run("pending block", func(t *testing.T) {
+		block0, err := mainnetGw.BlockByNumber(context.Background(), 0)
+		require.NoError(t, err)
+
+		blockID := rpc.BlockID{Pending: true}
+
+		mockReader.EXPECT().Pending().Return(blockchain.Pending{Block: block0}, nil)
+		mockReader.EXPECT().L1Head().Return(&core.L1Head{}, nil)
+
+		resp, rpcErr := handler.BlockWithReceipts(blockID)
+		header := resp.BlockHeader
+
+		var transactions []rpc.TransactionWithReceipt
+		for i, tx := range block0.Transactions {
+			receipt := block0.Receipts[i]
+			adaptedTx := rpc.AdaptTransaction(tx)
+
+			transactions = append(transactions, rpc.TransactionWithReceipt{
+				Transaction: adaptedTx,
+				Receipt:     rpc.AdaptReceipt(receipt, tx, rpc.TxnAcceptedOnL2, nil, 0),
+			})
+		}
+
+		assert.Nil(t, rpcErr)
+		assert.Equal(t, &rpc.BlockWithReceipts{
+			Status: rpc.BlockPending,
+			BlockHeader: rpc.BlockHeader{
+				Hash:             header.Hash,
+				ParentHash:       header.ParentHash,
+				Number:           header.Number,
+				NewRoot:          header.NewRoot,
+				Timestamp:        header.Timestamp,
+				SequencerAddress: header.SequencerAddress,
+				L1GasPrice:       header.L1GasPrice,
+				StarknetVersion:  header.StarknetVersion,
+			},
+			Transactions: transactions,
+		}, resp)
+	})
+
+	t.Run("accepted L1 block", func(t *testing.T) {
```

Comment:
I don't think we're returning execution resources correctly. For example block1 contains [builtin_instance_counter](https://github.com/NethermindEth/juno/blob/main/clients/feeder/testdata/mainnet/block/1.json#L128) which we don't seem to be returning

Reply:
At least when I print our output, I only get

```
"execution_resources": {
                    "steps": 29
                }
```

Reply:
The we still aren't returning all the data in Execution_resources for some reason. We are also not returning [data_availability](https://github.com/starkware-libs/starknet-specs/blob/v0.7.0-rc1/api/starknet_api_openrpc.json#L4037) from what I can see

Reply:
I'm inserting the following at the end of the "accepted L1 block" test, and comparing it against that's in the block by hand

```
qwe, _ := json.MarshalIndent(transactions, "", "")
fmt.Println(string(qwe))
panic(1)
```